app/policies/comment_policy.rb

```diff
@@ -0,0 +1,34 @@
+class CommentPolicy < ApplicationPolicy
+  alias_method :comment, :record
+
+  def index?
+    !comment.private? || comment.resource.member?(user.profile)
+  end
+
+  def create?
+    !comment.of_comment? &&
```

Comment:
in `Comment::CommentsController.create`, authorization is called on the parent.

```ruby
!comment.of_comment?
```

Is this really useful?